addons/l10n_fr_certification/models/res_company.py

```diff
@@ -2,41 +2,65 @@
         company = super(ResCompany, self).create(vals)
         #when creating a new french company, create the securisation sequence as well
-        if company.country_id == self.env.ref('base.fr'):
-            company._create_secure_sequence()
+        if self._is_accounting_unalterable():
+            sequence_fields = ['l10n_fr_secure_sequence_id', 'l10n_fr_closure_sequence_id']
+            company._create_secure_sequence(sequence_fields)
         return company
 
     @api.multi
     def write(self, vals):
         res = super(ResCompany, self).write(vals)
         #if country changed to fr, create the securisation sequence
-        if vals.get('country_id') and vals.get('country_id') == self.env.ref('base.fr').id:
-            self.filtered(lambda c: not c.l10n_fr_secure_sequence_id)._create_secure_sequence()
+        for company in self:
+            if company._is_accounting_unalterable():
+                sequence_fields = ['l10n_fr_secure_sequence_id', 'l10n_fr_closure_sequence_id']
+                company._create_secure_sequence(sequence_fields)
         return res
 
-    def _create_secure_sequence(self):
+    def _create_secure_sequence(self, sequence_fields):
         """This function creates a no_gap sequence on each companies in self that will ensure
         a unique number is given to all posted account.move in such a way that we can always
         find the previous move of a journal entry.
         """
         for company in self:
-            vals = {
-                'name': 'French Securisation of account_move_line - ' + company.name,
-                'code': 'FRSECUR',
-                'implementation': 'no_gap',
-                'prefix': '',
-                'suffix': '',
-                'padding': 0,
-                'company_id': company.id}
-            seq = self.env['ir.sequence'].create(vals)
-            company.write({'l10n_fr_secure_sequence_id': seq.id})
+            vals_write = {}
+            for seq_field in sequence_fields:
+                if not company[seq_field]:
+                    vals = {
+                        'name': 'French Securisation of ' + seq_field + ' - ' + company.name,
+                        'code': 'FRSECUR',
+                        'implementation': 'no_gap',
+                        'prefix': '',
+                        'suffix': '',
+                        'padding': 0,
+                        'company_id': company.id}
+                    seq = self.env['ir.sequence'].create(vals)
+                    vals_write[seq_field] = seq.id
+            if vals_write:
+                company.write(vals_write)
+
+    def _is_vat_french(self):
+        return self.vat and self.vat.startswith('FR') and len(self.vat) == 13
+
+    def _is_accounting_unalterable(self, raise_on_nocountry=False):
+        if raise_on_nocountry and not self.vat and not self.country_id:
+            raise UserError(_('Please set up a country or a VAT number on the company %s.') % self.name, )
```

Comment:
forget what I said about a new boolean field in res.company to store the fact the accounting should be unalterable for it: I forgot we're talking about a stable version, so no new field... but I still seriously don't believe that code is worth for people having their company not correctly configured: just put a line in the __manifest__.py of the module telling they have to set the country/vat on the company.

Reply:
The raise on no country set was a ProductOwner request

Reply:
@kebeclibre sometimes, we have to say 'No' to productowners... when their fantasies are just making us create crap on top of good code.

@fgi-odoo are you behind every questionable decision made in that PR? I do not buy that error in case the country/vat is not set on the company: it will silently return False instead